tensorboard/plugins/scalar/scalars_plugin.py

```diff
@@ -119,6 +121,26 @@ def scalars_impl(self, ctx, tag, run, experiment, output_format):
         else:
             return (values, "application/json")
 
+    def _scalars_multirun_impl(self, ctx, tag, runs, experiment):
+        """Result of the form `(body, mime_type)`."""
+        all_scalars = self._data_provider.read_scalars(
+            ctx,
+            experiment_id=experiment,
+            plugin_name=metadata.PLUGIN_NAME,
+            downsample=self._downsample_to,
+            run_tag_filter=provider.RunTagFilter(runs=runs, tags=[tag]),
+        )
+        result = {}
+        for run in all_scalars:
+            scalars = all_scalars[run][tag]
+            # Note we do not raise an error if data for a given run was not
+            # found; we just omit the run that case.
+            if scalars:
```

Comment:
What’s the purpose of this `if scalars` check? We know that `scalars` is
a list of `ScalarDatum` values (e.g., it can’t be `None`), so this omits
the run if it has empty data. But then a missing time series is
indistinguishable from a time series that is simply empty.

It’s (currently) true that in both the multiplexer data provider and
TensorBoard.dev we don’t keep track of empty time series, but there’s no
such restriction in the data provider model,\* so I don’t see why we
should go out of our way to lose information here. Can we just drop the
`if`-condition?

Sorry if I was unclear earlier; what I meant to convey is that the
structure of the data provider methods already has a simple, lossless
way to keep track of which time series exist and whether they have data,
and that plugin code should try to stick to that when possible. In most
cases this should generally amount to just doing the straightforward
thing without extra conditions.

\* If you want a motivating example, imagine some tuning service that
knows ahead of time what metrics you want to track.

Reply:
OK, I cleaned up this case-- thanks for clarifying.  I see now that what confused me is line 109 in the single-run case, which (per your other comment) suggests impossible states, e.g. the run is populated but the tag is not.  IIUC, that one would also be clearer as

```
if !all_scalars:
    raise NotFoundError(...)
scalars = all_scalars[run][tag]
```

(Not changing it here, though, so as not to pollute the PR.)